### ISSUE TITLE
setColumnDefs event leak

### ISSUE THREAD

**Issue by Llorx** [ag-grid/ag-grid#7777]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

When you call `setColumnDefs` with a new list of columns, `SelectAllFeature.prototype.setComp` is called, adding new events and not clearing the previous ones. I'm right now with a list in the thousands events assigned, which makes the interface go really slow:
![image](https://github.com/ag-grid/ag-grid/assets/702490/47bca800-6e02-441b-b5ef-999810d78edb)

ag-grid-community version `26.2.1` (the last one available for IE11, which is LTS):
![image](https://github.com/ag-grid/ag-grid/assets/702490/7d3658a9-eb6f-4d09-8669-f3b327dce979)

**Comment by kiril-matev** [comment]
Hi @Llorx,

Thanks for reaching out.

Please note that AG Grid v26 was designated as a long-term support version only for issues reported with the **environments that are discontinued** in v27 and only for the duration while these environments had not reached their end of life date. This means that only issues related to Polymer, Angular 7, AngularJS or Internet Explorer 11 in AG Grid 26 were in scope for this while these technologies were still supported by their vendors. Microsoft announced the end of life of Internet Explorer 11 on June 15th, 2022. This is why issues related to Internet Explorer 11 are no longer admissible for AG Grid 26 as Internet Explorer 11 is no longer a supported environment, as per Microsoft's announcement.

We're looking forward to receiving a bug report demonstrating this issue as per our bug submission template (pasted below). Please make sure to test with later versions of AG Grid as this issue may already be addressed. If that's the case, please upgrade to the AG Grid version which no longer has this issue - we recommend upgrading to the latest version.

**Providing a Reproducible Scenario**
Accepted reproducible scenarios are
- A description of the detailed steps to reproduce your behaviour in one of our examples in the docs.
- A plunker

If you decide to send us a plunker, from any example in our website use the plunker button in there to fork your own code by following the steps below:

- Select the framework that is appropriate to you from the drop-down at the top of the page
- Open it in plunker. (Use the button plunker in our example)
- Add your changes so that the behaviour is reproduced
- Save and Freeze the plunker(On the top left corner)
- Send us the link to the plunker(You can copy the URL from the browser)

**Please tell us about your environment:**


* **AG Grid version:** X.X.X


* **Browser:**


* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

We're looking forward to your response.

**Comment by Llorx** [comment]
Hello @kiril-matev thank you for answering.

Microsoft ended support for IE11 only in specific Windows versions, as stated in their website: https://learn.microsoft.com/en-us/lifecycle/faq/internet-explorer-microsoft-edge#what-is-the-lifecycle-policy-for-internet-explorer-

IE11 is still alive in multiple production systems, specially some Windows Servers I have access and they are still supported by Microsoft. I would like to recommend my clients to use the latest version of everything, including ag-grid, but that's impossible here 😞

I'm on my phone. I will setup a test when I reach a computer. Maybe the problem still persists even in the last versión.

**Comment by Llorx** [comment]
@kiril-matev

In the end it was the `refreshHeader` method. It leaks events as the headers are not destroyed (it seems). In this plnkr, each time you click the refresh header to redraw them, you see how the amount of events are increasing, which are also drawn on the header itself so you can easily check the increasing numbers: https://plnkr.co/edit/fu3EgOhIikQxKI3r

I would like to take into consideration that IE11 was not officially discontinued, as it was only discontinued on some Windows versions, not all Windows versions.

**Comment by kiril-matev** [comment]
Hi @Llorx,

Thanks for sending this plunker reproducing the issue - much appreciated.

I've confirmed this issue is fixed in v27.0 - see this in the modified version of your plunker using AG Grid v27.0: https://plnkr.co/edit/fjbOaeo5G4ZTeOTc

It's been 2 and a half years since the release of AG Grid v26.2.1. Designating it as LTS was a limited time, short-term measure to tackle issues for customers using Internet Explorer 11 to provide them additional time to plan a migration to a different web browser. It was never intended as a way provide support for AG Grid v26.2.1 indefinitely for users who aren't planning to move away from Internet Explorer 11. This is why we're not planning to release an update to v26.2.1 fixing this issue. 

Please upgrade your AG Grid version to v27.0 or later to resolve this issue.

**Comment by Llorx** [comment]
> Hi @Llorx,
> 
> Thanks for sending this plunker reproducing the issue - much appreciated.
> 
> I've confirmed this issue is fixed in v27.0 - see this in the modified version of your plunker using AG Grid v27.0: https://plnkr.co/edit/fjbOaeo5G4ZTeOTc
> 
> It's been 2 and a half years since the release of AG Grid v26.2.1. Designating it as LTS was a limited time, short-term measure to tackle issues for customers using Internet Explorer 11 to provide them additional time to plan a migration to a different web browser. It was never intended as a way provide support for AG Grid v26.2.1 indefinitely for users who aren't planning to move away from Internet Explorer 11. This is why we're not planning to release an update to v26.2.1 fixing this issue.
> 
> Please upgrade your AG Grid version to v27.0 or later to resolve this issue.

As I've stated, is impossible to remove IE11 in these installations as they are still supported by Microsoft so we don't have a word here. I guess we should search for another solution then.
